### PR TITLE
Add D symbol demangling

### PR DESCRIPTION
**Pull request by BorisCarvajal**
I copied many lines from Rust commit, I hope it works.

Issue #242

### CODE REVIEW

File: 3rdparty/perfparser.cmake [hunk 1]
```diff
@@ -46,6 +46,13 @@ if (RUSTC_DEMANGLE_INCLUDE_DIR AND RUSTC_DEMANGLE_LIBRARY)
     target_compile_definitions(hotspot-perfparser PRIVATE HAVE_RUSTC_DEMANGLE=1)
 endif()
 
+set(D_DEMANGLE_LIBRARY "" CACHE STRING "Path to the libddemangle.so library.")
```
[the remark this answers is not in the document]
Comment (BorisCarvajal) [reply 3rdparty/perfparser.cmake:49]: > does it not have an include library?

I can make a D function accessible from C++ using `extern (C++)` on D side, just need to link.

[the remark this answers is not in the document]
Comment (milianw) [reply 3rdparty/perfparser.cmake:49]: yes, please do that. ideally make it a separate github project similar to the rustc demangle library. ship a header there and also document how one would build the .so in the first place then

[the remark this answers is not in the document]
Comment (milianw) [reply 3rdparty/perfparser.cmake:49]: ping? any update on this?

**Comment by milianw** [comment]
if anyone else would like to work on this, please reference https://github.com/KDAB/hotspot/issues/269 and implement D demangling in a similar fashion, then nothing should be needed on our side (except for packaging purposes)

**Comment by milianw** [comment]
we got a different way implemented now thanks to @lievenhey , I'm closing this approach here now then

**Pull request closed without merging**